spec/03-plugins/35-azure-functions/01-access_spec.lua

```diff
@@ -21,6 +21,33 @@ for _, strategy in helpers.each_strategy() do
         protocols  = { "http", "https" },
       }
 
+      -- Mocking lua-resty-http's request_uri function
+      db.plugins:insert {
+        name = "pre-function",
+        route = { id = route2.id },
+        config = {
+          access = {
+            [[
+              local http = require "resty.http"
+              local json = require "cjson"
+              http.request_uri = function (self, uri, params)
+                return {
+                  status = 200,
+                  body = json.encode({
+                    status = 200,
+                    uri = uri,
+                    params = params,
+                  }),
+                  headers = {
+                    ["Content-Type"] = "application/json",
+                  }
+                }
+              end
+            ]]
+          }
+        }
+      }
+
```

Comment:
Not sure if this is the right way to mock. This completely bypasses the lua-resty-http library and we are not really testing the integration between `azure-functions` with lua-resty-http anymore. Effectively this file now became a unit test instead.

We should try to run this test while making sure `request_uri` is still really executing against a fake local service.

Reply:
Azure plugin does not support specify port which makes integration test kind of difficult, we'll need to use either 80 or 443 to serve local mock server

Reply:
Even if we do a patch of `http.request_uri` and change the default port, we have to make sure lua-resty-http still runs, otherwise this defeats the purpose of the test and is basically the same as disabling it.

Reply:
>  do a patch of http.request_uri and change the default port
That is exactly the same approach I've been thinking 😄 

I changed in that way and use a HTTP mock server.